### PR TITLE
feat: adding possibility to set default tags

### LINKED ISSUE #135: Default Tags
Add possibility to set default tags

### PR DESCRIPTION
**Pull request by vlastahajek**
Closes #135

Added possibility set default tags

- [X] CHANGELOG.md updated
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: api/write/options.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ type Options struct {
 	precision time.Duration
 	// Whether to use GZip compression in requests. Default false
 	useGZip bool
+	// Point settings, default tags
+	pointSettings *PointSettings
```
Comment (sranka) [inline api/write/options.go:29]: Wouldn't be better to simply use `defaultTags` herein. Precision (as a point setting) is also part of `Options`.

File: internal/write/service.go [hunk 2]
```diff
@@ -150,14 +151,67 @@ func (w *Service) WriteBatch(ctx context.Context, batch *Batch) error {
...
 	var buffer bytes.Buffer
 	e := lp.NewEncoder(&buffer)
 	e.SetFieldTypeSupport(lp.UintSupport)
 	e.FailOnFieldErr(true)
 	e.SetPrecision(w.writeOptions.Precision())
+	var m lp.Metric
 	for _, point := range points {
-		_, err := e.Encode(point)
+		if len(w.writeOptions.PointSettings().DefaultTags()) > 0 {
+			m = &pointWithDefaultTags{
```
Comment (sranka) [inline internal/write/service.go:207]: it might be better to reuse `pointToEncode` outside the for loop

File: internal/write/service_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	p.AddField("i", 1)
+
+	s, err = srv.EncodePoints(p)
+	require.Nil(t, err)
+	assert.Equal(t, "x,dt1=val1,xt=1,zdt=val2 i=1i\n", s)
+	assert.Len(t, p.TagList(), 1)
+
+	p = write.NewPointWithMeasurement("d")
+	p.AddTag("id", "1")
+	// do not overwrite point tag
```
Comment (sranka) [inline internal/write/service_test.go:43]: well done

File: options.go [hunk 4]
```diff
@@ -148,6 +148,14 @@ func (o *Options) HttpOptions() *http.Options {
 	return o.httpOptions
 }
 
+// AddDefaultTag adds a default tag. DefaultTags are added to each written point.
+// If a tag with the same key already exist it is overwritten.
+// If a point already defines a such tag, it is left unchanged
```
Comment (sranka) [inline options.go:153]: defines such a tag

File: api/write/options.go (hunk 1 above)
Comment (vlastahajek) [reply api/write/options.go:29]: I'm 100% for this. This was my first idea as well, but when I looked at Java and Js client I see default tags are part of PointSetting

Comment (sranka) [reply api/write/options.go]: PointSettngs serve different purpose in JS, it is not an API configuration iface.

**Pull request merged**